scaler/worker/agent/processor/processor.py

```diff
@@ -108,8 +110,14 @@ def __interrupt(self, *args):
 
     def __suspend(self, *args):
         assert self._resume_event is not None
+        assert self._resumed_event is not None
+
         self._resume_event.wait()  # stops any computation in the main thread until the event is triggered
 
+        # Ensures the processor agent knows we stopped waiting on `_resume_event`, as to avoid re-entrant wait on the
+        # event.
+        self._resumed_event.set()
+
```

Comment:
Using `Event` from the Spawn context didn't help with this issue. It still can create a re-entrant issue on the `wait()` method when the processor handles the `__suspended` signal twice in a short amount of time:

```
[...]
  File "/host/scaler/scaler/worker/agent/processor/processor.py", line 188, in __on_receive_object_content
    self.__process_task(task)
  File "/host/scaler/scaler/worker/agent/processor/processor.py", line 230, in __process_task
    result = function_with_logger(*args)
  File "/host/scaler/tests/test_nested_task.py", line 62, in fibonacci
    return a.result() + b.result()
  File "/host/scaler/scaler/client/future.py", line 104, in result
    self._result_ready_event.wait(timeout)
  File "/usr/lib/python3.10/threading.py", line 607, in wait
    signaled = self._cond.wait(timeout)
  File "/usr/lib/python3.10/threading.py", line 320, in wait
    waiter.acquire()
  File "/host/scaler/scaler/worker/agent/processor/processor.py", line 118, in __suspend
    self._resume_event.wait()  # stops any computation in the main thread until the event is triggered        <<<===== 1st signal
  File "/usr/lib/python3.10/multiprocessing/synchronize.py", line 349, in wait
    self._cond.wait(timeout)
  File "/usr/lib/python3.10/multiprocessing/synchronize.py", line 268, in wait
    self._lock.acquire()
  File "/host/scaler/scaler/worker/agent/processor/processor.py", line 118, in __suspend
    self._resume_event.wait()  # stops any computation in the main thread until the event is triggered        <<<===== 2nd signal received while still inside wait(). Lock the whole process on Linux.
```

The second `_resumed_event` prevents this.